compiler/passes/analyses/track_volume.py

```diff
@@ -26,190 +26,201 @@ def analyze(self, program: Program) -> dict: # The main function of the class
+                -1]  # Since volumes is a list, we wrap our single volume data in its own list. This is to avoid any issues when reading a disposed variable's entry down the line
+            self.variable_volume[instructions.defs['name']][
+                'size'] = 0  # A disposed variable doesn't have a presence on the board. It's size is therefore zero.
 
+    def _handle_dispose(self,
+                        instructions: dict):  # This is an internal simulation of the proper dispose function. It is used internally by mix and split when the host variable(s) are destroyed by the execution of the instruction
 
-def _handle_dispose(self, instructions : dict): # This is an internal simulation of the proper dispose function. It is used internally by mix and split when the host variable(s) are destroyed by the execution of the instruction
+        if instructions['offset'] >= 0:
+            if self.variable_volume.get(instructions['name'], None) is not None:
+                self.variable_volume[instructions['name']]['volumes'][instructions['offset']] = -1  # Since volumes is a list, we wrap our single volume data in its own list. This is to avoid any issues when reading a disposed variable's entry down the line
+            else:
+                self.violation_found = True
 
-    if (instructions['offset'] >=0):
-        if(self.variable_volume.get(instructions['name'], None) != None):
-            self.variable_volume[instructions['name']]['volumes'][instructions['offset']] = -1 # Since volumes is a list, we wrap our single volume data in its own list. This is to avoid any issues when reading a disposed variable's entry down the line
-            #self.variable_volume[instructions['name']]['size']    = 0;    # A disposed variable doesn't have a presence on the board. It's size is therefore zero.
         else:
-            self.violation_found = True;
+            self.variable_volume[instructions['name']]['volumes'] = [
+                -1]  # Since volumes is a list, we wrap our single volume data in its own list. This is to avoid any issues when reading a disposed variable's entry down the line
+            self.variable_volume[instructions['name']][
+                'size'] = 0  # A disposed variable doesn't have a presence on the board. It's size is therefore zero.
 
-    else:
-        self.variable_volume[instructions['name']]['volumes'] = [-1]; # Since volumes is a list, we wrap our single volume data in its own list. This is to avoid any issues when reading a disposed variable's entry down the line
-        self.variable_volume[instructions['name']]['size']    = 0;    # A disposed variable doesn't have a presence on the board. It's size is therefore zero.
+    def handle_mix(self, instructions: IR):
+        encoded_quantity = self._program.symbol_table.get_local(instructions.defs['name'], "main").value.volume['quantity'] # The quantity stored in an encoded float value.

```

Comment:
This uses the `value` attribute of a variable.  You should instead use your new `volumes` attribute, as `value` is updated to the _last parsed value_ with this variable name.

Reply:
I believe we discussed this and agreed that we could naively assume that a variable's volume is always equal to its lowest possible value.